prediction_market_agent_tooling/data_download/langfuse_data_downloader.py

```
@@ -0,0 +1,280 @@
+        raise ValueError(
+            f"Failed to fetch {market_type} market {market_id} resolution: {e}"
+        )
+
+
+def parse_date(date_str: str, param_name: str) -> DatetimeUTC:
+    try:
+        return DatetimeUTC.from_datetime(datetime.fromisoformat(date_str))
+    except ValueError:
+        typer.echo(f"Error: Invalid date format for {param_name}: {date_str}")
+        typer.echo("Expected format: YYYY-MM-DD or YYYY-MM-DDTHH:MM:SS")
+        raise typer.Exit(1)
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)